test/parallel/test-stream-pipe-cleanup-pause.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+var stream = require('stream');
+
+var reader = new stream.Readable();
+var writer1 = new stream.Writable();
+var writer2 = new stream.Writable();
+var writer1Writes = 0;
+var writer2Writes = 0;
+
+reader._read = function(n) {};
+
+writer1._write = function(chunk, encoding, cb) {
+  ++writer1Writes;
+  this.emit('chunk-received');
+  cb();
+};
+writer1.once('chunk-received', function() {
+  reader.unpipe(writer1);
+  reader.pipe(writer2);
+  reader.push(new Buffer(560000));
```

Comment:
It was copied from the test case in #2323. I think the point though is to fill up the internal buffer to cause the `write()` to return false.